n2v/grid/grider.py

```diff
@@ -0,0 +1,918 @@
+        natoms = len(indx)
+        #Atomic numbers and Atomic positions
+        zs = [mol_dict["elez"][i] for i in indx]
+        rs = [self.mol.geometry().np[i] for i in indx]
+
+        offset = 0
+        with np.errstate(divide='ignore'):
+            for i_block in blocks:
+                b_points = i_block.npoints()
+                offset += b_points
+                x = i_block.x().np
+                y = i_block.y().np
+                z = i_block.z().np
+                for atom in range(natoms):
+                    r =  np.sqrt( (x-rs[atom][0])**2 + (y-rs[atom][1])**2+ (z-rs[atom][2])**2)
+                    vext[offset - b_points : offset, 0] += -1.0 * zs[atom] / r
+                for i in range(len(vext[:,0])):
+                    if np.isinf(vext[i,0]) == True:
+                        vext[i,0] = 0.0
+            vext[:,1] = vext[:,0]
+
+        self.grid.vext = vext
+
+         ####################################
+
+        #HARTREE + ESP
+        esp_wfn = psi4.core.ESPPropCalc(self.wfn)
+        offset = 0
+        for i_block in blocks:
+            b_points = i_block.npoints()
+            offset += b_points
+            x = i_block.x().np[:,None]
+            y = i_block.y().np[:,None]
+            z = i_block.z().np[:,None]
+            xyz = np.concatenate((x,y,z), axis=1) 
+            grid_block = psi4.core.Matrix.from_array(xyz)
+            esp[offset - b_points : offset, 0] = esp_wfn.compute_esp_over_grid_in_memory(grid_block).np
+
+        #Hartree
+        hartree = - 1.0 * (vext + esp)
+        v_fa = -1.0 / (self.nalpha + self.nbeta) * hartree
+        self.grid.hartree = hartree
+        self.grid.v_fa = v_fa
+        self.grid.esp = esp
+
+        ####VXC################################
+        warnings.warn("Only LDA fucntionals are supported on the grid")
+        ingredients = {}
+        offset = 0
+        for i_block in blocks:
+            b_points = i_block.npoints()
+            offset += b_points
+            ingredients["rho"] = density[offset - b_points : offset, :]
+            if self.ref == 1:
+                functional = Functional(1, 1)
+            else:
+                functional = Functional(1, 2) 
+            xc_dictionary = functional.compute(ingredients)
+            vxc[offset - b_points : offset, :] = np.squeeze(xc_dictionary['vrho'])
+
+        self.grid.vxc = vxc
+        ####################################
+
+        #Generate 2D information
+        if grid_type == "cubic":
+            self.c_grid.density = np.reshape(density, (N0, N1, N2, density.shape[1]))
+            self.c_grid.vext = np.reshape(density, (N0, N1, N2, vext.shape[1]))
+            self.c_grid.hartree = np.reshape(density, (N0, N1, N2, hartree.shape[1]))
+            self.c_grid.esp = np.reshape(density, (N0, N1, N2, esp.shape[1]))
+            self.c_grid.vxc = np.reshape(density, (N0, N1, N2, vxc.shape[1]))
+                
+    def build_rectangular_grid(self, 
```

Comment:
Do we still need this function? I thought its functional is totally replaced by above functions.